index.d.ts

```diff
@@ -0,0 +1,85 @@
+declare module '@pusher/push-notifications-server' {
+    class PushNotifications {
+        constructor(options: PushNotifications.PushNotificationsOptions);
+        /**
+         *
+         * @param interests Array of interests to send the push notification to, ranging from 1 to 100 per publish request.
+         * @param publishRequest
+         */
+        publish<T extends PushNotifications.PublishRequestBase>(
+            interests: string[],
+            publishRequest: T['apns'] extends undefined
```

Comment:
Does this part mean "a publishRequest is either a `PublishRequestWithFcm` or a `PublishRequestWithApns`" ?  If so, it should be possible to publish to FCM and APNs at the same time

Reply:
It says if it has no apns, it _must_ have fcm, effectively enforcing an OR condition, but not XOR, because the interfaces both have an optional key of the other one in it